ragondin/routers/indexer.py

```diff
@@ -16,68 +15,58 @@
+            raise HTTPException(status_code=404, detail=f"File {file_id} already exists in partition {partition}")
         # Load metadata
+        metadata = metadata or "{}" 
         metadata = json.loads(metadata)
-        if metadata is None:
-            metadata = [{}] * len(files)
-        elif isinstance(metadata, list):
-            if len(metadata) != len(files):
-                raise HTTPException(status_code=400, detail="Number of metadata entries should match the number of files.")
-        elif isinstance(metadata, dict):
-            metadata = [metadata]
-        else:
-            raise HTTPException(status_code=400, detail="Metadata should be a dictionary or a list of dictionaries.")
-        # Load collection name from the request
-        if collection_name:
-            if not isinstance(collection_name, str):
-                raise HTTPException(status_code=400, detail="collection_name must be a string.")
+        if not isinstance(metadata, dict):
+            raise HTTPException(status_code=400, detail="Metadata should be a dictionary")
+            
+        # Add file_id to metadata
+        metadata["file_id"] = file_id
 
-        sub_dir = collection_name if collection_name else config.vectordb.default_collection_name
         # Create a temporary directory to store files
-        save_dir = Path(DATA_DIR) / sub_dir
+        save_dir = Path(DATA_DIR)
         save_dir.mkdir(parents=True, exist_ok=True)
         
-        # Save the uploaded files
-        for i, file in enumerate(files):
-            file_path = save_dir / Path(file.filename).name
-            logger.info(f"Processing file: {file.filename} and saving to {file_path}")
-            with open(file_path, "wb") as buffer:
-                buffer.write(await file.read())
-            # Now pass the file path to the Indexer
-            await indexer.add_files2vdb(path=file_path, metadata=metadata[i], collection_name=collection_name)
-
-        return JSONResponse(content={"message": f"Files processed and added to the vector database : {[file.filename for file in files]}"}, status_code=200)
+        # Save the uploaded file
+        file_path = save_dir / Path(file.filename).name
+        logger.info(f"Processing file: {file.filename} and saving to {file_path}")
+        with open(file_path, "wb") as buffer:
+            buffer.write(await file.read())
+        # Now pass the file path to the Indexer
+        await indexer.add_files2vdb(path=file_path, metadata=metadata, partition=partition)
+
+        return JSONResponse(content={"message": f"File processed and added to the vector database : {file_id}"}, status_code=200)
     except Exception as e:
         raise HTTPException(status_code=500, detail=str(e))
 
-@router.post("/search/", response_model=None)
-async def search(query_params: SearchRequest, indexer: Indexer = Depends(get_indexer)):
+@router.get("/", response_model=None)
+async def search_multiple_partitions(
```

Comment:
Semantically I don't think a search API should be a `GET`: for me it is more a `POST /{partitions}/search`  or `POST /search/multiple_partitions` (or `user_ids` rather than the `partition` term, see other comment)